src/ray/common/client_connection.h

```diff
@@ -22,13 +22,38 @@ namespace ray {
 ray::Status TcpConnect(boost::asio::ip::tcp::socket &socket,
                        const std::string &ip_address, int port);
 
+// Forward declaration for AsyncWriteCallbackHelper
+template <typename T>
+class ServerConnection;
+
+template <class T>
```

Comment:
Why create a new class for this?

Reply:
Talked with @guoyuhong offline, this could be just a method of ServerConnection